modules/openapi-generator/src/main/resources/python/model_templates/methods_shared.mustache

```diff
@@ -12,4 +12,21 @@
 
     def __getattr__(self, attr):
         """get the value of an attribute using dot notation: `instance.attr`"""
-        return self.{{#attrNoneIfUnset}}get{{/attrNoneIfUnset}}{{^attrNoneIfUnset}}__getitem__{{/attrNoneIfUnset}}(attr)
\ No newline at end of file
+        return self.{{#attrNoneIfUnset}}get{{/attrNoneIfUnset}}{{^attrNoneIfUnset}}__getitem__{{/attrNoneIfUnset}}(attr)
+
+    def __copy__(self):
+        cls = self.__class__
+        result = cls.__new__(cls)
```

Comment:
Is this copying a payload instantiated from client data `cls(**kwargs)`
Or one received from the server `cls._from_openapi_data(**kwargs)`?
If so should we not use __new__ and use one of the above options?

Reply:
that seems reasonble, ideally just _from_openapidata should be used, since we are copying the "data", and don't want the type checking (ie. checking for read-only would not be required).

Reply:
Ah so then because we are option2, will be not need the changes to the __init__ signature because that's a different code path?

Reply:
So how about changing this to use _from_openapi_data rather than __new__?

Reply:
agree, i will make that change.

Reply:
FYI: one can use the self._spec_property_naming to determine which code path to use when copying a self instance.